contracts/core/cross_chain_manager/logic/EthCrossChainManager.sol

```diff
@@ -11,12 +11,17 @@ import "./../interface/IEthCrossChainData.sol";
 contract EthCrossChainManager is IEthCrossChainManager, UpgradableECCM {
     using SafeMath for uint256;
 
+    address public whiteListContract;
```

Comment:
better to change it to constant instead of using storage (which is a pollution)

Reply:
eth-contract 实际上会部署在多个以太系的链上，不同链的lockproxy是不一致的，硬编码可能会导致不统一

Reply:
then use multiple versions of codes instead of writing it here

Reply:
for better future compatibility

Reply:
thats ok ：）

Reply:
lockproxy 通过api能动态获取吗？

Reply:
How are you supporting other already deployed LockProxys? eg LockProxyPip1?

Reply:
will add to whiteList

Reply:
But this only supports a single lockproxy contract, and data contract only supports a single ccm (owner).

Reply:
i'm considering using mapping to store whitelist info, later i'll release the new version

Reply:
Okay thank you.

Reply:
checkout the version

Reply:
Thanks. Btw can I understand why a whitelist instead of blacklist data contract is used? It seems since it's `msg.sender` that is checked for ownership, reentrant call is not possible.

Reply:
Actually, this version acts as an emergency fix , it will upgrade after . But  for now , whiteList is the safest solution . We are working on better solutions and will release it  when we sure they are safe.